File: jar-shell/src/main/java/com/beancounter/shell/auth/OAuth2Response.java
package com.beancounter.shell.auth;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class OAuth2Response {
  @JsonProperty("access_token")
  private String token;
  @JsonProperty("expires_in")
  private long expiry;
  @JsonProperty("refresh_token")
  private String refreshToken;
  @JsonProperty("token_type")
  private String type;
  private String scope;

}

File: jar-auth/src/main/kotlin/com/beancounter/auth/model/OAuth2Response.kt
package com.beancounter.auth.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Standard OAuth 2 response to a login request
 */
@JsonIgnoreProperties(ignoreUnknown = true)
class OAuth2Response(
    @JsonProperty("access_token") var token: String,
    @JsonProperty("scope") var scope: String,
    @JsonProperty("expires_in") var expiry: Long,
    @JsonProperty("token_type") var type: String,
    @JsonProperty("refresh_token") var refreshToken: String?
)
